spec/loophp/collection/CollectionSpec.php

```diff
@@ -27,13 +27,53 @@
 use PhpSpec\Exception\Example\MatcherException;
 use PhpSpec\ObjectBehavior;
 use stdClass;
+use TypeError;
 use function gettype;
 use const E_USER_DEPRECATED;
 use const INF;
 use const PHP_EOL;
 
 class CollectionSpec extends ObjectBehavior
 {
+    public function it_can_all(): void
+    {
+        $this::fromIterable([1, 2, 3])
+            ->all()
+            ->shouldIterateAs([1, 2, 3]);
+
+        $this::fromIterable(['foo' => 'f', 'bar' => 'b'])
+            ->all()
+            ->shouldIterateAs(['foo' => 'f', 'bar' => 'b']);
+
+        $duplicateKeyGen = static function (): Generator {
+            yield 'a' => 1;
+
+            yield 'b' => 2;
+
+            yield 'a' => 3;
+        };
+
+        $this::fromIterable($duplicateKeyGen())
+            ->shouldIterateAs($duplicateKeyGen());
+
+        $this::fromIterable($duplicateKeyGen())
+            ->all()
+            ->shouldIterateAs(['a' => 3, 'b' => 2]);
+
+        $nonArrayKeyGen = static function (): Generator {
+            yield ['a'] => 1;
+
+            yield ['b'] => 2;
+        };
+
+        $this::fromIterable($nonArrayKeyGen())
+            ->shouldIterateAs($nonArrayKeyGen());
+
+        $this::fromIterable($nonArrayKeyGen())
+            ->shouldThrow(TypeError::class)
+            ->during('all');
```

Comment:
I actually didn't realise this before but `all` cannot be used at all if you have keys that are not `int|string`, a type error will be thrown